client/src/templates/Challenges/utils/frame.ts

```diff
@@ -69,7 +74,7 @@ const createHeader = (id = mainPreviewId) => `
       }
       if (element && element.nodeName === 'A' && new URL(element.href).hash === '') {
         e.preventDefault();
-        window.parent.window.alert('Normally this link would bring you to another website! It works!' + ' This is a link to: ' + '(' + element.href + ')');
+        window.parent.window.alert("${alertText}" + "(" + element.href + ")");
```

Comment:
Thank you for working on this @Sboonny !
Is it possible to put the entire string into translations.json and use interpolation for `element.href` value? Like we do [here for the {{username}}](https://github.com/freeCodeCamp/freeCodeCamp/blob/8855d90854bf4fd1f54b1c156b755313592faa59/client/i18n/locales/english/translations.json#L476) for example.

In some languages (like Japanese) the order of words are different from English due to grammatical difference. It will make it easier for translators to make the translation look natural if they can change the order of the words in translation.

If it's impossible, it's okay. At least for Japanese, we can translate this string with this fixed order.

Reply:
I remember, trying to implement the link, because Arabic will face same issue. I will be honest, I don't remember what stopped me from implementing it, maybe I was tunnel visioned into making the string appear or an error that I didn't  document. 

Thank you for remind me about this, I am going to give it another try. if I run into an issue, I will inform you

Reply:
@sidemt I don't know, how I forgot this 🤣 but It's possible, but I amn't skilled enough to do it.

Note: 

`i18n` is the library we use to access the translation.
`window` is every global variable, it's accessible everywhere.
`element.href` is the clickable link, like `www.freecatappphoto.com`


But here is the jest of it. `window` doesn't read `i18n` and we need to inject `i18n` in it. Oliver advised me to render `i18n` translation outside, then push it into the `window`.

`element.href` is a variable that isn't accessible outside `window`, because it wouldn't exist until someone click on the link. So `i18n` can't read the variable it.

Here is two solution. but again I don't know how to do it. 

1- Making the process, an async function so `window` and `i18n` render at the same time.

2- Removing all of this for an alert component, that we are planning to make.

Reply:
About that... I did a little more thinking about this. There isn't a fundamental difference between what we do with `proxyLogger` and what we want to do here.  Both make calls to functions in the wider React app accessible within the frame.

Which is to say, sorry @Sboonny I mislead you!

I'll see what I can do and push a commit, if that's okay.

Reply:
> sorry @Sboonny I mislead you!

@ojeytonwilliams I don't really care, we did do something, I didn't know that we could do.

> I'll see what I can do and push a commit, if that's okay.

Feel free to, I don't mind 👍.

meanwhile I will look at `proxyLogger`, as well.

Thank you for helping us out 💚

Reply:
Too late 🤣, I now know that I can do it so it will be done 👍.

Reply:
16 days, whoops. Well it works now perfectly 😅